pkgs/swiftgen/swift2objc/lib/src/parser/parsers/relations_parser.dart

```diff
@@ -0,0 +1,22 @@
+import '../../ast/declarations/compounds/class_declaration.dart';
+import '../_core/type_defs.dart';
+
+class RelationsParser {
+  final JsonMap symbolgraphJson;
+  final DeclarationsMap declarations;
+
+  RelationsParser(this.symbolgraphJson, this.declarations);
+
+  void wireUp() {
+    final List relations = symbolgraphJson["relationships"];
+    for (final relation in relations) {
+      String methodId = relation["source"];
+      String classId = relation["target"];
+
+      final methodDeclaraion = declarations[methodId] as ClassMethodDeclaration;
```

Comment:
What if you pull it out of the map as a `Declaration`, then have an `addDeclaration` method to `ClassDeclaration` that decides whether to add it to `methods` or `properties` depending on which type of declaration it is?

You don't have to do this now, since you only have methods ATM, but you'll need to do something like that when you add other declaration types.